nemo/collections/asr/parts/submodules/multi_head_attention.py

```diff
@@ -453,7 +456,11 @@ def forward(self, query, key, value, pad_mask, pos_emb, cache=None, cache_next=N
 
                 out[is_index_global_attn_nonzero] += out_global_to_all
 
-        return self.linear_out(out.reshape(n_batch, -1, self.h * self.d_k)[:, :T])
+        ret = self.linear_out(out.reshape(n_batch, -1, self.h * self.d_k)[:, :T])
+        if cache is None:
```

Comment:
Do we need it here as ret looks like to be always single?

Reply:
Ret is always single, yes - but cache comes from update_cache here, not from ret.